demisto_sdk/commands/common/docker_helper.py

```diff
@@ -235,13 +239,21 @@ def create_container(
         """
         Creates a container and pushing requested files to the container.
         """
-        container: docker.models.containers.Container = (
-            init_global_docker_client().containers.create(
-                image=image, command=command, environment=environment, **kwargs
+        docker_client = init_global_docker_client()
+
+        try:
+            container: docker.models.containers.Container = (
+                docker_client.containers.create(
+                    image=image, command=command, environment=environment, **kwargs
+                )
             )
-        )
+        except DockerException as e:
+            container.remove()
+            raise e
```

Comment:
why raising? don't we want to remove the container and re-create it?

Reply:
We are already recreating the container with the wrapper, here we just want to remove the container if it has already been created and for some reason it fails

Reply:
but the retry wrapper doesn't catch `DockerException` so retry will not be re-triggered here.